In [68]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
from IPython.display import display
import scipy.stats as stats
import matplotlib.pyplot as plt

In [113]:
gdpcomp_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/gdpandcomp.dta"
gdpcomp_df = pd.read_stata(gdpcomp_file_path)

indshares_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/indshares.dta"
indshares_df = pd.read_stata(indshares_file_path)

otheroutcomes_bpl_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/otheroutcomes_bpl.dta"
otheroutcomes_bpl_df = pd.read_stata(otheroutcomes_bpl_file_path)

spred10_bpl_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/spred10_bpl.dta"
spred10_bpl_df = pd.read_stata(spred10_bpl_file_path)

spred0_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/spred0.dta"
spred0_df = pd.read_stata(spred0_file_path)

spred10_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/spred10.dta"
spred10_df = pd.read_stata(spred10_file_path)

spred20_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/spred20.dta"
spred20_df = pd.read_stata(spred20_file_path)

spred30_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/spred30.dta"
spred30_df = pd.read_stata(spred30_file_path)

spred40_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/spred40.dta"
spred40_df = pd.read_stata(spred40_file_path)

ACS_100sampled_file_path = "/Users/markoshea/Desktop/Working Directories/Datasets/rep_files/DATA/ACS_100sampled.csv"
ACS_100sampled_df = pd.read_csv(ACS_100sampled_file_path, low_memory=False)

In [116]:
ACS_100sampled_df

year    sample     serial   cbserial    hhwt       cluster statefip  \
0       1930   1930 1%        1.0        NaN  100.95  1.930000e+12  Alabama   
1       1930   1930 1%       24.0        NaN  100.95  1.930000e+12  Alabama   
2       1930   1930 1%       47.0        NaN  100.95  1.930000e+12  Alabama   
3       1930   1930 1%       72.0        NaN  100.95  1.930000e+12  Alabama   
4       1930   1930 1%       94.0        NaN  100.95  1.930000e+12  Alabama   
...      ...       ...        ...        ...     ...           ...      ...   
554553  2011  2011 ACS  1366786.0  1388057.0   85.00  2.011014e+12  Wyoming   
554554  2011  2011 ACS  1366836.0  1419022.0  363.00  2.011014e+12  Wyoming   
554555  2011  2011 ACS  1366879.0  1447426.0  206.00  2.011014e+12  Wyoming   
554556  2011  2011 ACS  1366919.0  1469976.0   25.00  2.011014e+12  Wyoming   
554557  2011  2011 ACS  1366965.0  1498458.0   85.00  2.011014e+12  Wyoming   

             strata                                gq  pernum  ...  \
0       110100100.0  Households under 1970 definition       1  ...   
1       110100100.0  Households under 1970 definition       2  ...   
2       110100101.0  Households under 1970 definition       1  ...   
3       120100500.0  Households under 1970 definition       6  ...   
4       120100501.0  Households under 1970 definition       3  ...   
...             ...                               ...     ...  ...   
554553      40056.0  Households under 1970 definition       1  ...   
554554      30056.0  Households under 1970 definition       1  ...   
554555      40056.0  Households under 1970 definition       2  ...   
554556      20056.0  Households under 1970 definition       7  ...   
554557      20056.0  Households under 1970 definition       1  ...   

                       educ                        educd             empstat  \
0                       NaN                          NaN            Employed   
1                       NaN                          NaN  Not in labor force   
2                       NaN                          NaN            Employed   
3                       NaN                          NaN  Not in labor force   
4                       NaN                          NaN  Not in labor force   
...                     ...                          ...                 ...   
554553             Grade 12  Regular high school diploma            Employed   
554554   4 years of college            Bachelor's degree            Employed   
554555  N/A or no schooling       No schooling completed                 NaN   
554556  N/A or no schooling                          NaN                 NaN   
554557             Grade 12  Regular high school diploma  Not in labor force   

                  empstatd                                 ind1990 wkswork1  \
0                  At work                                     NaN      NaN   
1       Not in Labor Force                                     NaN      NaN   
2                  At work                                     NaN      NaN   
3       Not in Labor Force                                     NaN      NaN   
4       Not in Labor Force                                     NaN      NaN   
...                    ...                                     ...      ...   
554553             At work              General government, n.e.c.      NaN   
554554             At work  Miscellaneous manufacturing industries      NaN   
554555                 NaN                    N/A (not applicable)      NaN   
554556                 NaN                    N/A (not applicable)      NaN   
554557  Not in Labor Force              Eating and drinking places      NaN   

           wkswork2 uhrswork  incwage  migplac1  
0               NaN      NaN      NaN       NaN  
1               NaN      NaN      NaN       NaN  
2               NaN      NaN      NaN       NaN  
3               NaN      NaN      NaN       NaN  
4               NaN      NaN      NaN       NaN  
...     

Explanation of column names for `ACS_100sampled_df`:
- **year** - Year survey was conducted  
- **sample** - Specific ACS sample identifier  
- **serial** - Unique household identifier within sample  
- **cbserial** - Census Bureau household identifier for internal use  
- **hhwt** - Household weight, used to produce population estimates  
- **cluster** - Geographic cluster or primary sampling unit for statistical weighting  
- **statefip** - Federal Information Processing Standard (FIPS) state code  
- **strata** - Survey stratum for variance estimation  
- **gq** - Indicates whether the person lives in a household (`1`) or in group quarters (`2` for institutions, `3` for non-institutional quarters like dorms)  
- **pernum** - Person number within the household (`1` householder, `2` spouse, etc.)  
- **perwt** - Survey weight for individuals, used to adjust sample representation  
- **sex** - Respondent's sex (`1 = Male`, `2 = Female`)  
- **age** - Respondent's age in years  
- **birthyr** - Respondent's birth year  
- **educ** - Education level in broad categories  
- **educd** - Detailed education category  
- **empstat** - Employment status  
- **empstatd** - More detailed employment status code  
- **ind1990** - Industry of employment based on 1990 Census industry codes  
- **wkswork1** - Number of weeks worked last year (broad categories)  
- **wkswork2** - Number of weeks worked last year (exact count)  
- **uhrswork** - Usual hours worked per week  
- **incwage** - Wage and salary income in nominal dollars  
- **migplac1** - Place of residence 1 year ago (coded by state or country) 

In [115]:
print(ACS_100sampled_df.dtypes)

year          int64
sample       object
serial      float64
cbserial    float64
hhwt        float64
cluster     float64
statefip     object
strata      float64
gq           object
pernum        int64
perwt       float64
sex          object
age          object
birthyr       int64
educ         object
educd        object
empstat      object
empstatd     object
ind1990      object
wkswork1    float64
wkswork2     object
uhrswork     object
incwage     float64
migplac1     object
dtype: object


In [ ]:
type(ACS_100sampled_df['statefip'].iloc[0])

str

In [ ]:
type(ACS_100sampled_df['gq'].iloc[0])

str

In [ ]:
type(ACS_100sampled_df['sex'].iloc[0])

str

In [ ]:
type(ACS_100sampled_df['birthyr'].iloc[0])

numpy.int64

In [ ]:
type(ACS_100sampled_df['statefip'].iloc[0])

str

In [ ]:
type(ACS_100sampled_df['empstat'].iloc[0])

str

In [ ]:
type(ACS_100sampled_df['empstatd'].iloc[0])

str

In [ ]:
ACS_100sampled_df['statefip'].apply(type).value_counts()

statefip
<class 'str'>    554558
Name: count, dtype: int64

In [ ]:
print(ACS_100sampled_df['year'].value_counts())

year
2000    140814
1990    125011
1980    113431
2011     31120
2010     30617
2009     30308
1970     20304
1950     19222
1960     17999
1940     13517
1930     12215
Name: count, dtype: int64


In [ ]:
print(ACS_100sampled_df['statefip'].value_counts())

statefip
California              60188
New York                41148
Texas                   37980
Pennsylvania            27542
Florida                 27026
Illinois                26231
Ohio                    24403
Michigan                20651
New Jersey              17263
North Carolina          15475
Georgia                 14944
Massachusetts           13689
Virginia                13399
Indiana                 12706
Missouri                11724
Tennessee               11242
Wisconsin               11225
Washington              10714
Maryland                10262
Minnesota               10046
Louisiana                9267
Alabama                  9250
Kentucky                 8554
Arizona                  8167
South Carolina           7859
Colorado                 7590
Oklahoma                 7167
Connecticut              7068
Iowa                     6631
Oregon                   6375
Mississippi              6092
Kansas                   5686
Arkansas                 5530
W

In [ ]:
print(ACS_100sampled_df['sample'].value_counts())

sample
2000 5%              140814
1990 5%              125011
1980 5%              113431
2011 ACS              31120
2010 ACS              30617
2009 ACS              30308
1970 Form 1 State     20304
1950 1%               19222
1960 1%               17999
1940 1%               13517
1930 1%               12215
Name: count, dtype: int64


In [ ]:
print(ACS_100sampled_df['serial'].value_counts())

serial
65848.0      5
204946.0     5
512676.0     4
47880.0      4
106706.0     4
            ..
4588612.0    1
4588572.0    1
4588491.0    1
4588442.0    1
1366965.0    1
Name: count, Length: 523979, dtype: int64


In [ ]:
print(ACS_100sampled_df['hhwt'].value_counts())

hhwt
20.0      122234
100.0      43680
21.0       17169
18.0       17130
22.0       14179
           ...  
1027.0         1
1162.0         1
686.0          1
520.0          1
487.0          1
Name: count, Length: 642, dtype: int64


In [ ]:
#essentially functions like id apparently
print(ACS_100sampled_df['cluster'].value_counts())

cluster
1.930000e+12    1
2.000022e+12    1
2.000022e+12    1
2.000022e+12    1
2.000022e+12    1
               ..
1.980042e+12    1
1.980042e+12    1
1.980042e+12    1
1.980042e+12    1
2.011014e+12    1
Name: count, Length: 554558, dtype: int64


In [ ]:
print(ACS_100sampled_df['strata'].value_counts())

strata
6.000000e+01    50528
4.800000e+01    41145
4.000000e+00    21499
6.400000e+01    19466
2.200000e+01    19448
                ...  
8.012221e+10        1
1.972940e+05        1
1.983040e+05        1
8.011221e+10        1
3.109800e+05        1
Name: count, Length: 42173, dtype: int64


In [ ]:
print(ACS_100sampled_df['gq'].value_counts())

gq
Households under 1970 definition               538848
Other group quarters                             7689
Group quarters--Institutions                     6910
Additional households under 1990 definition      1024
Fragment                                           57
Additional households under 2000 definition        30
Name: count, dtype: int64


In [ ]:
print(ACS_100sampled_df['pernum'].value_counts())

pernum
1     211517
2     152327
3      88787
4      55123
5      25690
6      11178
7       5084
8       2255
9       1198
10       661
11       341
12       202
13        78
14        44
15        34
16        15
18         6
21         5
20         4
17         3
23         2
19         2
25         1
22         1
Name: count, dtype: int64


In [ ]:
print(ACS_100sampled_df['perwt'].value_counts())

perwt
20.0      122008
100.0      43709
18.0       16993
21.0       16158
19.0       13187
           ...  
545.0          1
680.0          1
828.0          1
1118.0         1
636.0          1
Name: count, Length: 691, dtype: int64


In [ ]:
print(ACS_100sampled_df['sex'].value_counts())

sex
Female    284737
Male      269821
Name: count, dtype: int64


In [ ]:
print(ACS_100sampled_df['educ'].value_counts())

educ
Grade 12                     142082
Grade 5, 6, 7, or 8           68849
N/A or no schooling           60260
Nursery school to grade 4     53814
1 year of college             52793
4 years of college            44987
5+ years of college           26807
2 years of college            24267
Grade 10                      23488
Grade 11                      21183
Grade 9                       20378
3 years of college             3435
Name: count, dtype: int64


In [ ]:
print(ACS_100sampled_df['empstat'].value_counts())

empstat
Employed              243865
Not in labor force    165231
Unemployed             17501
Name: count, dtype: int64


In [ ]:
ind1990_valcts = ACS_100sampled_df['ind1990'].value_counts()
print(ind1990_valcts)

ind1990
N/A (not applicable)                               215199
All construction                                    19615
Elementary and secondary schools                    17254
Eating and drinking places                          15994
DID NOT RESPOND                                     12318
                                                    ...  
Leather tanning and finishing                          47
Watches, clocks, and clockwork operated devices        42
Coast Guard                                            35
Mobile home dealers                                    32
Pipe lines, except natural gas                         29
Name: count, Length: 246, dtype: int64


In [ ]:
print(ACS_100sampled_df['wkswork1'].value_counts())

wkswork1
0.0     206607
52.0    115564
50.0     10909
40.0      8356
48.0      7009
12.0      4189
20.0      3919
30.0      3649
36.0      3530
26.0      3320
51.0      2611
32.0      2377
16.0      2362
45.0      2267
24.0      2214
8.0       2188
10.0      2012
44.0      1862
49.0      1807
42.0      1736
46.0      1712
25.0      1644
4.0       1625
28.0      1519
6.0       1376
35.0      1318
2.0       1066
38.0      1065
15.0      1044
47.0       889
3.0        838
5.0        747
18.0       720
1.0        716
14.0       679
22.0       606
39.0       597
13.0       590
34.0       530
37.0       450
9.0        440
43.0       439
17.0       360
7.0        352
41.0       333
27.0       310
21.0       275
11.0       260
23.0       248
33.0       230
29.0       194
19.0       168
31.0       167
Name: count, dtype: int64


In [ ]:
print(ACS_100sampled_df['wkswork2'].value_counts())

wkswork2
50-52 weeks    172405
40-47 weeks     22046
1-13 weeks      21959
14-26 weeks     21643
27-39 weeks     20559
48-49 weeks     10860
Name: count, dtype: int64


In [ ]:
print(ACS_100sampled_df['uhrswork'].value_counts())

uhrswork
40    113496
50     17109
45     11546
20     11356
30     10383
       ...  
94         5
87         4
79         4
89         3
93         1
Name: count, Length: 99, dtype: int64


In [ ]:
print(ACS_100sampled_df['incwage'].value_counts())

incwage
0.0         155616
30000.0       4668
20000.0       4478
25000.0       3716
12000.0       3649
             ...  
4710.0           1
24941.0          1
13470.0          1
22977.0          1
272000.0         1
Name: count, Length: 14666, dtype: int64


In [98]:
spred0_df

year   stfips   pred0num0    pred0num1     pred0num2     pred0num3  \
0    1909  Alabama  592732.000  471526.0000  393640.00000  259808.00000   
1    1919  Alabama  608774.875  535768.0000  403451.00000  293007.15625   
2    1929  Alabama  636388.750  601056.3125  467701.34375  330611.25000   
3    1939  Alabama  613650.000  609095.0000  507279.00000  403222.00000   
4    1949  Alabama  704427.000  565100.0000  479429.00000  430512.00000   
..    ...      ...         ...          ...           ...           ...   
552  1969  Wyoming   65100.000   66600.0000   45700.00000   35300.00000   
553  1979  Wyoming   87140.000   78540.0000  101500.00000   69820.00000   
554  1989  Wyoming   76098.000   72615.0000   61218.00000   81944.00000   
555  1999  Wyoming   64653.000   78367.0000   65767.00000   65798.00000   
556  2009  Wyoming  230689.000  220258.0000  242086.00000  209958.00000   

         pred0num4      pred0num5      pred0num6     pred0num7     pred0num8  \
0    170280.000000  125148.000000   72109.000000  32487.000000   7725.000000   
1    233605.453125  133204.453125   87888.398438  39353.878906  11047.179688   
2    271555.500000  182618.546875   90754.046875  46033.199219  14334.899414   
3    288501.000000  207765.000000  136664.000000  51070.000000  16317.000000   
4    357804.000000  243109.000000  181695.000000  85156.000000  21510.000000   
..             ...            ...            ...           ...           ...   
552   38600.000000   34700.000000   24600.000000  15900.000000   4900.000000   
553   43340.000000   39840.000000   29520.000000  15640.000000   5980.000000   
554   57961.000000   38125.000000   33151.000000  21583.000000   8955.000000   
555   82598.000000   59595.000000   34821.000000  27393.000000  12534.000000   
556  215167.000000  254510.000000  164964.000000  84180.000000  45463.000000   

       pred0num9    pred0all  
0    1001.000000  2126456.00  
1    1959.900024  2348060.25  
2    1413.299927  2642467.00  
3    2579.000000  2836142.00  
4    2750.000000  3071492.00  
..           ...         ...  
552   600.000000   332000.00  
553  1040.000000   472360.00  
554  1539.000000   453189.00  
555  2323.000000   493849.00  
556  9613.000000  1676888.00  

[557 rows x 13 columns]

In [99]:
print(spred0_df.dtypes)

year         category
stfips       category
pred0num0     float32
pred0num1     float32
pred0num2     float32
pred0num3     float32
pred0num4     float32
pred0num5     float32
pred0num6     float32
pred0num7     float32
pred0num8     float32
pred0num9     float32
pred0all      float32
dtype: object


Explanation of column names for `spred0_df`:
- **year** – The specific year the data pertains to.  
- **stfips** – The **State FIPS Code**, a unique two-digit numerical identifier assigned to each U.S. state and territory.  
- **pred0num0** – The total weighted number of people **aged 0-9** in a given state-year.  
- **pred0num1** – The total weighted number of people **aged 10-19** in a given state-year.  
- **pred0num2** – The total weighted number of people **aged 20-29** in a given state-year.  
- **pred0num3** – The total weighted number of people **aged 30-39** in a given state-year.
- **pred0num4** – The total weighted number of people **aged 40-49** in a given state-year.    
- **pred0num5** – The total weighted number of people **aged 50-59** in a given state-year.  
- **pred0num6** – The total weighted number of people **aged 60-69** in a given state-year.  
- **pred0num7** – The total weighted number of people **aged 70-79** in a given state-year.  
- **pred0num8** – The total weighted number of people **aged 80-89** in a given state-year.  
- **pred0num9** – The total weighted number of people **aged 90+** in a given state-year.  
- **pred0all** – The total weighted population across **all age groups** in a given state-year.  

Note that spred10_df through spred40_df follow the same structure as spred0_df in column names and data types. Note that these are lagged so tht spred10.dta uses age structures from year - 10, spred20.dta uses from year - 20, and so on.

In [100]:
spred10_df

year   stfips     pred10num1     pred10num2     pred10num3  \
0    1909  Alabama  585514.625000  475631.250000  361204.687500   
1    1919  Alabama  624777.937500  551667.312500  403153.093750   
2    1929  Alabama  630614.062500  569343.000000  438423.093750   
3    1939  Alabama  635996.625000  609385.437500  520268.906250   
4    1949  Alabama  717758.812500  559893.812500  487697.906250   
..    ...      ...            ...            ...            ...   
552  1969  Wyoming   68921.937500   67428.929688   49119.144531   
553  1979  Wyoming   90429.578125   79071.296875  105248.703125   
554  1989  Wyoming   84826.859375   80066.257812   66677.296875   
555  1999  Wyoming   69256.554688   82967.632812   69295.281250   
556  2009  Wyoming       0.000000       0.000000       0.000000   

        pred10num4     pred10num5     pred10num6     pred10num7    pred10num8  \
0    235596.015625  142190.437500   93483.500000   41196.300781  10507.333984   
1    280060.625000  201806.718750  105099.570312   53081.058594  13601.478516   
2    303692.531250  229721.796875  144955.906250   55840.828125  15719.131836   
3    404853.312500  268814.250000  178646.468750   90359.695312  20662.902344   
4    419747.218750  332616.375000  203780.093750  122381.890625  32355.273438   
..             ...            ...            ...            ...           ...   
552   35614.238281   37772.496094   31126.798828   18188.546875   7349.518555   
553   69798.335938   41910.062500   35463.785156   22033.177734   7957.147949   
554   83754.953125   56493.292969   35303.464844   26155.691406  12013.445312   
555   66264.539062   80096.343750   56890.628906   28794.392578  15875.021484   
556       0.000000       0.000000       0.000000       0.000000      0.000000   

      pred10num9     pred10all  
0    1198.336914  1.361008e+06  
1    1731.952881  1.610202e+06  
2    1916.249146  1.759612e+06  
3    2432.446533  2.095423e+06  
4    2955.300781  2.161428e+06  
..           ...           ...  
552  1015.731506  2.476154e+05  
553  1346.731445  3.628292e+05  
554  2136.564453  3.626010e+05  
555  3211.578125  4.033954e+05  
556     0.000000  0.000000e+00  

[557 rows x 12 columns]

In [ ]:
spred20_df

In [ ]:
spred30_df

Explanation of column names for `spred10_bpl.dta`:

- **year** – The specific year the data pertains to.  
- **stfips** – The State FIPS Code, a unique two-digit numerical identifier assigned to each U.S. state and territory.  
- **pred10num1** – The total weighted number of people **aged 10-19** in a given state-year **based on their previous residence (`migplac1`)**.  
- **pred10num2** – The total weighted number of people **aged 20-29** in a given state-year **based on their previous residence**.  
- **pred10num3** – The total weighted number of people **aged 30-39** in a given state-year **based on their previous residence**.  
- **pred10num4** – The total weighted number of people **aged 40-49** in a given state-year **based on their previous residence**.  
- **pred10num5** – The total weighted number of people **aged 50-59** in a given state-year **based on their previous residence**.  
- **pred10num6** – The total weighted number of people **aged 60-69** in a given state-year **based on their previous residence**.  
- **pred10num7** – The total weighted number of people **aged 70-79** in a given state-year **based on their previous residence**.  
- **pred10num8** – The total weighted number of people **aged 80-89** in a given state-year **based on their previous residence**.  
- **pred10num9** – The total weighted number of people **aged 90+** in a given state-year **based on their previous residence**.  
- **pred10all** – The **total weighted population** across all age groups in a given state-year, based on individuals' previous residence.  

In [102]:
print(spred10_bpl_df.dtypes)

year          category
stfips        category
pred10num1     float32
pred10num2     float32
pred10num3     float32
pred10num4     float32
pred10num5     float32
pred10num6     float32
pred10num7     float32
pred10num8     float32
pred10num9     float32
pred10all      float32
dtype: object


In [103]:
otheroutcomes_bpl_df

year         stfips    poploc          earn           age2  \
0    1999        Alabama  187069.0  5.817407e+10  613166.000000   
1    2009        Alabama  907040.0  7.530679e+10  643299.000000   
2    1999         Alaska   24204.0  1.050963e+10   84830.000000   
3    2009         Alaska  158889.0  1.702236e+10  121460.666667   
4    1999        Arizona  231472.0  7.357752e+10  735670.000000   
..    ...            ...       ...           ...            ...   
97   2009  West Virginia  388206.0  2.727130e+10  228129.000000   
98   1999      Wisconsin  190513.0  8.535540e+10  689014.000000   
99   2009      Wisconsin  929323.0  1.094912e+11  763121.000000   
100  1999        Wyoming   13143.0  6.583203e+09   65767.000000   
101  2009        Wyoming   73959.0  1.099808e+10   80104.666667   

              age3           age4           age5           age6  \
0    646174.000000  657905.000000  506400.000000  358763.000000   
1    592984.000000  653923.333333  649360.666667  483142.666667   
2    101764.000000  114001.000000   68161.000000   28965.000000   
3    100089.333333  109808.666667  107655.000000   59269.666667   
4    767315.000000  714058.000000  533157.000000  392063.000000   
..             ...            ...            ...            ...   
97   227029.000000  250484.000000  279289.000000  215181.333333   
98   806081.000000  839041.000000  587808.000000  390646.000000   
99   688339.000000  822197.000000  825634.333333  543749.666667   
100   65798.000000   82598.000000   59595.000000   34821.000000   
101   69226.000000   72372.000000   85726.000000   55770.000000   

              age7  ...          work        tothrs       tothrs2    work_1  \
0    266242.000000  ...  2.112515e+06  3.964440e+09  3.967632e+09  0.822135   
1    285056.333333  ...  2.176369e+06  3.995055e+09  0.000000e+00  0.786460   
2     15967.000000  ...  3.376030e+05  6.115678e+08  6.123156e+08  0.877879   
3     24873.333333  ...  4.117103e+05  7.589605e+08  0.000000e+00  0.854132   
4    319105.000000  ...  2.452022e+06  4.588466e+09  4.590817e+09  0.825570   
..             ...  ...           ...           ...           ...       ...   
97   127266.333333  ...  8.226040e+05  1.518204e+09  0.000000e+00  0.763727   
98   319387.000000  ...  2.880923e+06  5.453806e+09  5.460429e+09  0.911555   
99   315624.333333  ...  3.019775e+06  5.492714e+09  0.000000e+00  0.876412   
100   27393.000000  ...  2.629720e+05  4.958444e+08  4.959175e+08  0.897541   
101   28353.666667  ...  3.089337e+05  5.894268e+08  0.000000e+00  0.898737   

       work_2    work_3     wage_1     wage_2     wage_3  nocollegedegree  
0    0.769441  0.235339  12.513549  16.692030  16.411985         0.820616  
1    0.735904  0.262070  15.494834  21.212828  21.705198         0.790537  
2    0.855414  0.347776  14.340936  19.960657  17.918724         0.772584  
3    0.840108  0.417152  18.879530  25.659828  23.191452         0.752796  
4    0.807066  0.234716  13.886732  18.312510  17.363073         0.779731  
..        ...       ...        ...        ...        ...              ...  
97   0.711863  0.232535  15.388000  19.812794  18.987354         0.827766  
98   0.883469  0.282562  13.723597  17.517805  16.010962         0.786454  
99   0.851011  0.315698  16.698347  22.343899  20.902395         0.750900  
100  0.855464  0.301618  11.267901  15.079386  12.999712         0.803739  
101  0.860896  0.376321  16.431173  20.613758  18.737406         0.768949  

[102 rows x 23 columns]

Explanation of column names for `otheroutcomes_bpl.dta`:

- **year** – The specific year the data pertains to.  
- **stfips** – The State FIPS Code, a unique two-digit numerical identifier assigned to each U.S. state and territory.  
- **poploc** – The total population in the state-year (based on past residence, `migplac1`).  
- **earn** – The total sum of **wage earnings** in the state-year.  
- **age2** – The number of people **aged 20-29** in the state-year.  
- **age3** – The number of people **aged 30-39** in the state-year.  
- **age4** – The number of people **aged 40-49** in the state-year.  
- **age5** – The number of people **aged 50-59** in the state-year.  
- **age6** – The number of people **aged 60-69** in the state-year.  
- **age7** – The number of people **aged 70-79** in the state-year.  
- **age8** – The number of people **aged 80-89** in the state-year.  
- **age9** – The number of people **aged 90+** in the state-year.  
- **cnn** – The total number of individuals in the dataset for a given state-year. 
- **work** – The total number of individuals who are **working** in the state-year.  
- **tothrs** – The total **predicted** number of hours worked in the state-year (based on `wkswork1p`).  
- **tothrs2** – The total number of hours worked in the state-year using the **original** work weeks variable (`wkswork1`).  
- **work_1** – The **average workforce participation rate** for people **under 40** in the state-year.  
- **work_2** – The **average workforce participation rate** for people **aged 40-59** in the state-year.  
- **work_3** – The **average workforce participation rate** for people **aged 60+** in the state-year.  
- **wage_1** – The **average wage** for people **under 40** in the state-year.  
- **wage_2** – The **average wage** for people **aged 40-59** in the state-year.  
- **wage_3** – The **average wage** for people **aged 60+** in the state-year.  
- **nocollegedegree** – The proportion of people **without a college degree** in the state-year.  

In [105]:
indshares_df

stfips  year      IND1      IND2      IND3      IND4      IND5  \
0    Alabama  1949  0.172709  0.024749  0.045725  0.180276  0.054233   
1    Alabama  1959  0.111314  0.011004  0.073814  0.265247  0.042504   
2    Alabama  1969  0.040129  0.007196  0.065498  0.262638  0.040775   
3    Alabama  1979  0.027247  0.011601  0.069292  0.259738  0.037241   
4    Alabama  1989  0.023440  0.007343  0.073146  0.231570  0.039340   
..       ...   ...       ...       ...       ...       ...       ...   
350  Wyoming  1969  0.086842  0.094737  0.062281  0.060526  0.057018   
351  Wyoming  1979  0.056466  0.138755  0.099004  0.058181  0.052716   
352  Wyoming  1989  0.060362  0.078945  0.079741  0.060259  0.052370   
353  Wyoming  1999  0.044245  0.068983  0.089407  0.056485  0.052944   
354  Wyoming  2009  0.040538  0.085036  0.083952  0.055691  0.047894   

         IND6      IND7      IND8      IND9     IND10     IND11     IND12  \
0    0.017108  0.022326  0.107970  0.018488  0.016899  0.087342  0.004175   
1    0.023402  0.025715  0.123332  0.026607  0.015810  0.096337  0.004304   
2    0.026937  0.035701  0.128875  0.031919  0.019742  0.072509  0.004244   
3    0.035799  0.041242  0.128664  0.045570  0.028050  0.037821  0.004938   
4    0.034689  0.041357  0.132605  0.050093  0.039904  0.030652  0.007298   
..        ...       ...       ...       ...       ...       ...       ...   
350  0.029825  0.029825  0.150877  0.029825  0.015789  0.061404  0.006140   
351  0.041466  0.026572  0.142505  0.041359  0.027965  0.038359  0.006000   
352  0.036262  0.024170  0.150066  0.041073  0.035229  0.035170  0.011895   
353  0.032799  0.023262  0.136927  0.045476  0.041123  0.039703  0.010389   
354  0.031975  0.019605  0.144099  0.039280  0.043405  0.030440  0.012516   

        IND13     IND14  
0    0.080683  0.164974  
1    0.094697  0.052093  
2    0.140590  0.058579  
3    0.192647  0.067790  
4    0.222857  0.065707  
..        ...       ...  
350  0.168421  0.067544  
351  0.199186  0.060966  
352  0.255335  0.079123  
353  0.273378  0.084878  
354  0.291900  0.073669  

[355 rows x 16 columns]

From `create.do`:

- **IND1** → `ind1990` between **2 and 32** (`1 < ind1990 < 33`)  
- **IND2** → `ind1990` between **40 and 50** (`39 < ind1990 < 51`)  
- **IND3** → `ind1990` is **exactly 60** (`ind1990 == 60`)  
- **IND4** → `ind1990` between **62 and 399** (`61 < ind1990 < 400`)  
- **IND5** → `ind1990` between **400 and 432** (`399 < ind1990 < 433`)  
- **IND6** → `ind1990` between **440 and 472** (`439 < ind1990 <= 472`)  
- **IND7** → `ind1990` between **500 and 571** (`499 < ind1990 < 572`)  
- **IND8** → `ind1990` between **580 and 691** (`579 < ind1990 < 692`)  
- **IND9** → `ind1990` between **700 and 712** (`699 < ind1990 < 713`)  
- **IND10** → `ind1990` between **714 and 760** (`713 < ind1990 < 761`)  
- **IND11** → `ind1990` between **761 and 791** (`760 < ind1990 < 792`)  
- **IND12** → `ind1990` between **800 and 810** (`799 < ind1990 < 811`)  
- **IND13** → `ind1990` between **812 and 893** (`811 < ind1990 < 894`)  
- **IND14** → `ind1990` greater than **893** (`893 < ind1990`)  

Therefore, from the 1990 Census Industry Codes (`ind1990`):

- **IND1** - Agriculture, Forestry, Fishing, Hunting  
- **IND2** - Mining (includes Oil and Gas Extraction)  
- **IND3** - Utilities  
- **IND4** - Construction and Manufacturing  
- **IND5** - Wholesale Trade  
- **IND6** - Retail Trade  
- **IND7** - Transportation and Warehousing  
- **IND8** - Information  
- **IND9** - Finance and Insurance  
- **IND10** - Real Estate and Rental Leasing  
- **IND11** - Professional, Scientific, and Technical Services  
- **IND12** - Educational Services  
- **IND13** - Health Care and Social Assistance  
- **IND14** - Arts, Entertainment, Recreation, Accommodation, and Food Services  

IND1 through IND14 represent the share of employment or economic activity attributed to specific industry sectors within each state and year. Source: https://www.census.gov/content/dam/Census/about/about-the-bureau/adrm/data-linkage/HUDMetadata/HUD-MTO/IndustryOccupationCodes/1990%20Census_SIC%20codes.pdf

In [106]:
indshares_stfips_valcts = indshares_df['stfips'].value_counts()
print(indshares_stfips_valcts)

stfips
Alabama                 7
Pennsylvania            7
Nevada                  7
New Hampshire           7
New Jersey              7
New Mexico              7
New York                7
North Carolina          7
North Dakota            7
Ohio                    7
Oklahoma                7
Oregon                  7
Rhode Island            7
Montana                 7
South Carolina          7
South Dakota            7
Tennessee               7
Texas                   7
Utah                    7
Vermont                 7
Virginia                7
Washington              7
West Virginia           7
Wisconsin               7
Nebraska                7
Missouri                7
Illinois                7
Mississippi             7
Arizona                 7
Arkansas                7
California              7
Colorado                7
Connecticut             7
Delaware                7
District of Columbia    7
Florida                 7
Georgia                 7
Idaho                   7
Wyomi

In [ ]:
gdpcomp_df

     stfips  year  indcode     comp       gdp
0       1.0  1969        1   7036.0   11866.0
1       1.0  1979        1  20220.0   33712.0
2       1.0  1989        1  40405.0   68394.0
3       1.0  1999      101  66951.0  112079.0
4       1.0  2009      101  96502.0  166315.0
..      ...   ...      ...      ...       ...
250    56.0  1969        1    796.0    1801.0
251    56.0  1979        1   3534.0    8063.0
252    56.0  1989        1   4793.0   11437.0
253    56.0  1999      101   7786.0   15728.0
254    56.0  2009      101  15360.0   34432.0

[255 rows x 5 columns]


In [108]:
print(gdpcomp_df.dtypes)

stfips     float32
year         int16
indcode      int16
comp       float64
gdp        float64
dtype: object


Explanation of column names for `gdpcomp_df`:

- **stfips** - Federal Information Processing Standards (FIPS) code, a unique two-digit numerical identifier assigned to each U.S. state and territory.  
- **year** - The specific year the data pertains to.  
- **indcode** - Denotes the industry code, which classifies industries according to NAICS (North American Industry Classification System).  
- **comp** – **Total Compensation (millions of dollars)** paid to employees in the state-year.  
- **gdp** – **Gross Domestic Product (GDP, millions of dollars)** for the state-year. 

- **stfips** – The **State FIPS Code**, a unique numerical identifier assigned to each U.S. state and territory.  
- **year** – The specific year the data pertains to (**1969, 1979, 1989, 1999, or 2009**).  
- **indcode** – The **Industry Code** from BEA’s classification system:  
  - **`1` (early years) and `101` (later years)** likely correspond to **total private sector or aggregate GDP/compensation data**.  
- **comp** – **Total employee compensation** (wages, salaries, and benefits) for the given state-year-industry.  
- **gdp** – **Gross Domestic Product (GDP) in current dollars**, measuring the total economic output for the state-year-industry.  

In [109]:
gdpcomp_indcode_valcts = gdpcomp_df['indcode'].value_counts()
print(gdpcomp_indcode_valcts)

indcode
1      153
101    102
Name: count, dtype: int64
